wpilibj/src/main/java/edu/wpi/first/wpilibj/shuffleboard/ContainerHelper.java

```diff
@@ -0,0 +1,114 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package edu.wpi.first.wpilibj.shuffleboard;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+import edu.wpi.first.networktables.NetworkTableType;
+import edu.wpi.first.wpilibj.Sendable;
+
+import static edu.wpi.first.networktables.NetworkTableType.kBoolean;
+import static edu.wpi.first.networktables.NetworkTableType.kBooleanArray;
+import static edu.wpi.first.networktables.NetworkTableType.kDouble;
+import static edu.wpi.first.networktables.NetworkTableType.kDoubleArray;
+import static edu.wpi.first.networktables.NetworkTableType.kRaw;
+import static edu.wpi.first.networktables.NetworkTableType.kString;
+import static edu.wpi.first.networktables.NetworkTableType.kStringArray;
+
+/**
+ * A helper class for Shuffleboard containers to handle common child operations.
+ */
+final class ContainerHelper {
+
+  private final ShuffleboardContainer m_container;
+  private final Set<String> m_usedTitles = new HashSet<>();
+  private final List<ShuffleboardComponent<?>> m_components = new ArrayList<>();
+  private final Map<String, ShuffleboardLayout> m_layouts = new LinkedHashMap<>(); // NOPMD
+
+  ContainerHelper(ShuffleboardContainer container) {
+    m_container = container;
+  }
+
+  List<ShuffleboardComponent<?>> getComponents() {
+    return m_components;
+  }
+
+  ShuffleboardLayout getLayout(String type, String title) {
+    if (!m_layouts.containsKey(title)) {
+      ShuffleboardLayout layout = new ShuffleboardLayout(m_container, type, title);
+      m_components.add(layout);
+      m_layouts.put(title, layout);
+    }
+    return m_layouts.get(title);
+  }
+
+  ComplexWidget add(String title, Sendable sendable) {
+    checkTitle(title);
+    ComplexWidget widget = new ComplexWidget(m_container, title, sendable);
+    m_components.add(widget);
+    return widget;
+  }
+
+  ComplexWidget add(Sendable sendable) throws IllegalArgumentException {
+    if (sendable.getName() == null || sendable.getName().isEmpty()) {
+      throw new IllegalArgumentException("Sendable must have a name");
+    }
```

Comment:
What we currently do in LiveWindow is silently ignore components without names.  If add() is only going to be called during initial configuration and never in the middle of robot code, it might be okay to throw, but is that the contract?